tools/eval/datasets/otb100.py

```diff
@@ -0,0 +1,522 @@
+
+    def __len__(self):
+        return len(self.videos)
+
+    def __iter__(self):
+        sorted_keys = sorted(list(self.videos.keys()))
+        for key in sorted_keys:
+            yield self.videos[key]
+
+def get_axis_aligned_bbox(region):
+    """Converts a region to (cx, cy, w, h) representing an axis-aligned box."""
+    nv = region.size
+    if nv == 8:
+        cx = np.mean(region[0::2])
+        cy = np.mean(region[1::2])
+        x1 = min(region[0::2])
+        x2 = max(region[0::2])
+        y1 = min(region[1::2])
+        y2 = max(region[1::2])
+        A1 = np.linalg.norm(region[0:2] - region[2:4]) * np.linalg.norm(region[2:4] - region[4:6])
+        A2 = (x2 - x1) * (y2 - y1)
+        s = np.sqrt(A1 / A2)
+        w = s * (x2 - x1) + 1
+        h = s * (y2 - y1) + 1
+    else:
+        x, y, w, h = region
+        cx = x + w / 2
+        cy = y + h / 2
+    return cx, cy, w, h
+
+def load_text_numpy(path, delimiter, dtype):
+    for d in delimiter:
+        try:
+            ground_truth_rect = np.loadtxt(path, delimiter=d, dtype=dtype)
+            return ground_truth_rect
+        except:
+            pass
+
+    raise Exception('Could not read file {}'.format(path))
+
+class OTB100:
+    def __init__(self, root):
+        # Go up one if directory is provided
+        root = os.path.abspath(root)
+        if root.endswith("OTB100"):
+            root = os.path.dirname(root)
+
+        self.dataset = OTBDATASET(root)
+
+    @property
+    def name(self):
+        return self.__class__.__name__
+
+    def eval(self, model):
+        for video in tqdm(self.dataset, desc="Evaluating: ", total=100, ncols=100):
+            total_time = 0
+            pred_bboxes = []
+            scores = []
+            track_times = []
+
+            for idx, (img, gt_bbox) in enumerate(video):
+                img = cv.cvtColor(img, cv.COLOR_BGR2RGB)
+                tic = cv.getTickCount()
+
+                if idx == 0:
+                    cx, cy, w, h = get_axis_aligned_bbox(np.array(gt_bbox))
+                    gt_bbox_ = (int(cx - w / 2), int(cy - h / 2), int(w), int(h))
+                    model.init(img, gt_bbox_)
+                    pred_bbox = gt_bbox_
+                    scores.append(None)
+                else:
+                    isLocated, bbox, score = model.infer(img)
+                    pred_bbox = bbox
+                    scores.append(score)
+
+                pred_bboxes.append(pred_bbox)
+                toc = (cv.getTickCount() - tic) / cv.getTickFrequency()
+                total_time += toc
+                track_times.append(toc)
+
+            model_path = os.path.join('OTB_results')
+            os.makedirs(model_path, exist_ok=True)
+            result_path = os.path.join(model_path, '{}.txt'.format(video.name))
+            with open(result_path, 'w') as f:
+                for bbox in pred_bboxes:
+                    f.write(','.join(map(str, bbox)) + '\n')
```

Comment:
We can drop result saving for now as they are not needed after evaluation is finished and we do not load them for the second time evaluation.

Reply:
Replaced with a global dict. If this isn't preferred, I can find another way tomorrow. :+1:

Reply:
Have you pulled recently? With the current branch, I am able to run the full evaluation with `python3 eval.py -m vittrack -d otb100 -dr OTB100`. I have no extra files. Worth noting that I do get different results than you, presumably because I don't have the patch properly applied locally.
![image](https://github.com/opencv/opencv_zoo/assets/43348991/67cbea9f-bc6f-4160-b47e-70e962f4aab5)

Reply:
Which opencv version are you using? I can confirm that both my latest built-from-source version and latest release of opencv do not work. Checking back to the last commit works. Did you do some modifications to the dataset or the opencv code?

Reply:
Using the dataset zip that you uploaded. I was able to run it using version 4.9.0.
![image](https://github.com/opencv/opencv_zoo/assets/43348991/f1908412-9c1f-4fde-b895-07ceff731cd9)

I am able to:
1. Create a new folder
2. `git clone git@github.com:ryan1288/opencv_zoo.git`
3. `git checkout ryan-vittracker_eval`
4. Move the new downloadable dataset into the `tools/eval` folder and unzip
5. `cd tools/eval`
6. `python3 eval.py -m vittrack -d otb100 -dr OTB100`
And it runs without problems.
![image](https://github.com/opencv/opencv_zoo/assets/43348991/426cfa53-dcc3-4c51-b0e7-ccf4d89e77e6)

I'll check my OpenCV installation to ensure there aren't discrepancies tomorrow. You're significantly more likely to have set it up correctly than me :+1: